lib/ConvoyFollower/src/App.cpp

```diff
@@ -227,25 +227,35 @@ void App::loop()
 
 void App::setErrorState()
 {
-    m_systemStateMachine.setState(&ErrorState::getInstance());
+    if (&ErrorState::getInstance() != m_systemStateMachine.getState())
+    {
+        m_systemStateMachine.setState(&ErrorState::getInstance());
+        m_v2vCommManager.triggerEmergencyStop();
+    }
 }
 
 void App::systemStatusCallback(SMPChannelPayload::Status status)
 {
-    switch (status)
+    if (m_lastRUStatus != status)
     {
-    case SMPChannelPayload::STATUS_FLAG_OK:
-        /* Nothing to do. All good. */
-        break;
+        /* Save last status. */
+        m_lastRUStatus = status;
 
-    case SMPChannelPayload::STATUS_FLAG_ERROR:
-        LOG_DEBUG("RU Status ERROR.");
-        setErrorState();
-        m_statusTimeoutTimer.stop();
-        break;
+        switch (status)
+        {
+        case SMPChannelPayload::STATUS_FLAG_OK:
+            /* Nothing to do. All good. */

```

Comment:
If there is only one good case and all others are errors, you can use a simple if/else